File: backend/java-reactive/src/main/java/com/beautysalon/reactive/model/Staff.java
package com.beautysalon.reactive.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.Table;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Table("staff")
public record Staff(
    @PrimaryKey @Id UUID id,
    @NotBlank(message = "Name is required") String name,
    @Email(message = "Invalid email format") String email,
    @Pattern(regexp = "\\+?[1-9]\\d{1,14}", message = "Invalid phone number") String phone,
    @NotBlank(message = "Role is required") String role,
    List<String> specialties,
    Boolean active,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
    public Staff {
        if (id == null) {
            id = UUID.randomUUID();
        }
        if (active == null) {
            active = true;
        }
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (updatedAt == null) {
            updatedAt = LocalDateTime.now();
        }
    }

    public static Staff create(String name, String email, String phone, String role, List<String> specialties) {
        return new Staff(null, name, email, phone, role, specialties, null, null, null);
    }

    public Staff withUpdatedFields(String name, String email, String phone, String role, 
                                 List<String> specialties, Boolean active) {
        return new Staff(
            this.id,
            name != null ? name : this.name,
            email != null ? email : this.email,
            phone != null ? phone : this.phone,
            role != null ? role : this.role,
            specialties != null ? specialties : this.specialties,
            active != null ? active : this.active,
            this.createdAt,
            LocalDateTime.now()
        );
    }
}

File: backend/dotnet/BeautySalonAPI/Models/Staff.cs
using System.ComponentModel.DataAnnotations;

namespace BeautySalonAPI.Models;

public class Staff
{
    [Key]
    public Guid Id { get; set; }

    [Required(ErrorMessage = "Name is required")]
    [StringLength(100, ErrorMessage = "Name cannot exceed 100 characters")]
    public string Name { get; set; } = string.Empty;

    [Required(ErrorMessage = "Email is required")]
    [EmailAddress(ErrorMessage = "Invalid email format")]
    public string Email { get; set; } = string.Empty;

    [Phone(ErrorMessage = "Invalid phone number")]
    [StringLength(20, ErrorMessage = "Phone number cannot exceed 20 characters")]
    public string Phone { get; set; } = string.Empty;

    [Required(ErrorMessage = "Role is required")]
    [StringLength(50, ErrorMessage = "Role cannot exceed 50 characters")]
    public string Role { get; set; } = string.Empty;

    public List<string>? Specialties { get; set; }

    public bool Active { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public Staff()
    {
        Id = Guid.NewGuid();
        Active = true;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = DateTime.UtcNow;
    }

    public static Staff Create(string name, string email, string phone, string role, List<string>? specialties = null)
    {
        return new Staff
        {
            Name = name,
            Email = email,
            Phone = phone,
            Role = role,
            Specialties = specialties
        };
    }

    public Staff Update(string? name = null, string? email = null, string? phone = null,
                       string? role = null, List<string>? specialties = null, bool? active = null)
    {
        return new Staff
        {
            Id = this.Id,
            Name = name ?? this.Name,
            Email = email ?? this.Email,
            Phone = phone ?? this.Phone,
            Role = role ?? this.Role,
            Specialties = specialties ?? this.Specialties,
            Active = active ?? this.Active,
            CreatedAt = this.CreatedAt,
            UpdatedAt = DateTime.UtcNow
        };
    }
}
